src/streaming/controllers/BufferController.js

```diff
@@ -559,6 +559,24 @@ function BufferController(config) {
             if (currentTimeRequest) {
                 rangeStart = Math.max(currentTimeRequest.startTime + currentTimeRequest.duration, rangeStart);
             }
+
+            // Never remove the contiguous range of targetTime in order to avoid flushes & reenqueues in WebKit based browsers
+            const ua = navigator.userAgent.toLowerCase();
+            if (/webkit/.test(ua) && !/chrome/.test(ua) && !/windows phone/.test(ua)) {
```

Comment:
I assume this is not a problem on MacOS Safari. We need to make sure that this only targets embedded devices. In any case the fix should be disabled by default, see settings comment above.

Reply:
With the computing power available on a desktop system like MacOS, this shouldn't be a problem. However, the issue here is that we can't just rely on the user agent to distinguish a WPE browser running on a set-top-box from Safari running on MacOS, since it's very common for deploys to use a fake user agent to avoid problems with third party websites. The [development reference platform used by Metrological/Comcast](https://github.com/WebPlatformForEmbedded/buildroot/blob/main/package/wpe/wpeframework-plugins/Config.in#L1341) is an example of such behaviour.

With this in mind, I like a lot your suggestion of having a `buffer.enableEmbeddedWebkitPruning` setting flag, because that would still allow to match Safari-like user agents (or even remove user-agent matching completely) and only enable the desired behaviour when the embedder webpage demands it. What do you think about this approach?

Reply:
Yes we can do that but disable the workaround by default. Applications/developers can then enable it via the settings.

Reply:
In the end I chose `enableCurrentTimeRangePruning`, set to `true` by default. A new version of the patch is available for review. Thank you!

Reply:
Please disable `enableCurrentTimeRangePruning` by default. The current logic is verified on multiple devices. The workaround you provide here is only required on specific platforms and should be disabled by default.

Reply:
I see that you run the new logic when the settings flag is disabled. In find that confusing, in my opinion we should run this new logic if the corresponding flag is set to `true`

Reply:
I decided to name the flag with a "positive" ("enable") name because I found no other settings flag with a "negative" name. If you prefer a flag that is disabled by default, then I would need to rename it to something like "avoidCurrentTimeRangePruning", false by default (traditional behaviour) and true on demand in order to enable the new behaviour.

Reply:
A new version of the patch with "avoidCurrentTimeRangePruning" (false by default) is available for review. I've also improved the log line, since reporting about an inverse pruning range (eg: [43.410666, 42.910666]) might seem confusing to the user. Now, when there's an inverse range, a clarification is added to explain that no actual range will be pruned.

Thanks!